### PR TITLE
WIP: escape hatches proposal

### PR DESCRIPTION
**Pull request by campionfellin**
## WIP

I based a large part of this proposal off of https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html

I also remember previously (maybe I'm misremembering) that you could do escape hatches in `cdk` with raw YAML, which is why I included it here.

If it's easier to read this rendered out in markdown, I can put it on the readme for my fork as well.

Resolves https://github.com/awslabs/cdk8s/issues/144

Signed-off-by: campionfellin <campionfellin@gmail.com>

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,6 +66,80 @@ Choose your weapon:
...
+  spec: {
+    replicas: 3,
+    podSpecTemplate: {
+      containers: [ container ]
+    }
+  },
+});
+```
+
+`cdk8s+` lets you set set replicas and the pod spec, but does not have an option for `strategy`. It leaves it to the default `RollingUpdate`. If you want to specify this, but still want to use `cdk8s+` constructs, you can use an override like so:
```
Comment (eladb) [inline README.md:96]:
```suggestion
At the moment, `cdk8s+` lets you define replicas and the pod spec, but does not have an option for `strategy`. It leaves it to the default `RollingUpdate`. If you want to specify this, but still want to use `cdk8s+` constructs, you can use an override like so:
```

File: README.md [hunk 2]
```diff
@@ -66,6 +66,80 @@ Choose your weapon:
...
+
+```ts
+import { Construct } from 'constructs';
+import { Chart, Raw } from 'cdk8s';
+
+class MyChart extends Chart {
+ constructor(scope: Construct, name: string) {
+    super(scope, name);
+
+    new Raw(this, 'raw', {
```
Comment (eladb) [inline README.md:114]: What's the difference between `Raw` and `Include`? I think they should be the same thing, no?

File: README.md [hunk 3]
```diff
@@ -66,6 +66,80 @@ Choose your weapon:
...
+      containers: [ container ]
+    }
+  },
+});
+```
+
+`cdk8s+` lets you set set replicas and the pod spec, but does not have an option for `strategy`. It leaves it to the default `RollingUpdate`. If you want to specify this, but still want to use `cdk8s+` constructs, you can use an override like so:
+
+```ts
+deployment.addOverride('spec.strategy', 'Recreate')
```
Comment (eladb) [inline README.md:99]: Mention the format of the first argument (is is a JSON query?)

Comment (eladb) [inline README.md:99]: Similarly to the AWS CDK, I think overrides should be implemented in the L1 layer, not L2.

I am not sure we need explicit overrides in L2 (cdk8s+). Users should be able to use `node.defaultChild` to find the L1 behind an L2 and then we should add overrides to the L1 layer.

File: README.md (hunk 2 above)
Comment (campionfellin) [reply README.md:114]: No difference, good callout! I actually didn't realize we had `Include` already :)

File: README.md (hunk 3 above)
Comment (campionfellin) [reply README.md:99]: Ah, will do that. Yes, intended to be a json path

Comment (campionfellin) [reply README.md:99]: Aha gotcha, I'll take another look here then

**Pull request closed without merging**